File: TP06/teclado.c
#include <stdio.h>
#include <string.h>

int main() {
    char linha[1001]; 

    while (fgets(linha, sizeof(linha), stdin) != NULL) {
        int abriu = 0;
        char dentro[1001] = "";
        char normal[1001] = "";
        int dentro_idx = 0;
        int normal_idx = 0;

        linha[strcspn(linha, "\n")] = 0;

        for (int i = 0; i < strlen(linha); i++) {
            if (linha[i] == '[') {
                abriu++;
            }
            if (linha[i] == ']') {
                abriu--;
            }
            if (abriu > 0 && linha[i] != '[') {
                dentro[dentro_idx++] = linha[i];
            }
            if (abriu == 0 && (linha[i] != '[' && linha[i] != ']')) {
                normal[normal_idx++] = linha[i];
            }
        }
        dentro[dentro_idx] = '\0';
        normal[normal_idx] = '\0';

        if (strlen(dentro) > 1) {
            printf("%s%s\n", dentro, normal);
        } else {
            printf("%s\n", normal);
        }
    }

    return 0;
}

File: TP06/teclado.java
import java.util.Scanner;
public class teclado{
	public static void main(String[]args){
		Scanner sc= new Scanner(System.in);
		while(sc.hasNextLine()){
			String linha=sc.nextLine();
			int abriu=0;
			String dentro="";
			String normal="";
			for(int i=0;i<linha.length();i++){
				if(linha.charAt(i)=='['){
					abriu++;
				}
				if(linha.charAt(i)==']'){
					abriu--;
				}
				if(abriu>0&&linha.charAt(i)!='['){ 
					dentro+=linha.charAt(i);
				}
				if(abriu==0&&(linha.charAt(i)!='['&&linha.charAt(i)!=']')){
					normal+=linha.charAt(i);
				}
			}
			if(dentro.length()>1){
				System.out.println(dentro+normal);
		}
		else{
			System.out.println(normal);
		}
		}
		sc.close();
	}
}


				
